api/v1alpha1/virtualmachineimage_conversion.go

```diff
@@ -390,6 +391,14 @@ func (src *VirtualMachineImage) ConvertTo(dstRaw ctrlconversion.Hub) error {
 		return err
 	}
 
+	// Manually restore data.
+	restored := &vmopv1.VirtualMachineImage{}
+	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
+		return err
+	}
+
+	dst.Status.DiskInfo = restored.Status.DiskInfo
```

Comment:
Why not just?

```golang
dst.Status = restored.Status
```

Reply:
Wouldn't that overwrite what was done in `convert_v1alpha1_VirtualMachineImageSpec_To_v1alpha3_VirtualMachineImageStatus` on line 385?

Reply:
When the`Status.Type` was recently added, we decided on punting doing any version conversions for that field: TKG images have a huge plaintext yaml, so this will effectively double their size for down-version users. And CCI already has slowness issues with images. And with the {Cluster}VirtualMachineImages we - VMOP controller - are the only one that does updates on these types so the saved version in the Annotation won't ever be used.

Reply:
Yep. We decided to narrow this to a given field.